src/engine/Resources/Sound.ts

```diff
@@ -274,8 +276,15 @@ export class Sound implements ILoadable, IAudio {
 
    /**
     * Play the sound, returns a promise that resolves when the sound is done playing
+    * Valid string inputs include:
+    * -'linearUp'
+    * -'linearDown'
+    * -'ease'
+    * -'ease-in'
+    * -'ease-out'
+    * -'ease-in-out'
     */
-   public play(volume?: string): Promise<boolean> {
+   public play(data?: string | number | IBezierPoints): Promise<boolean> {
```

Comment:
I like idea of simply saying `play(linearUp)` in sound. I probably would want to control the maximum volume somehow, I could imagine this getting loud accidentally.

Using a type union we can strongly type these options, take a look at events
```typescript
// Events.ts
export type kill = 'kill';

export type predraw = 'predraw';
export type postdraw = 'postdraw';

export type predebugdraw = 'predebugdraw';
export type postdebugdraw = 'postdebugdraw';

export type preupdate = 'preupdate';
export type postupdate = 'postupdate';

export type preframe = 'preframe';
export type postframe = 'postframe';

export type collision = 'collision';

export type initialize = 'initialize';
export type activate = 'activate';
...
// Actor.ts
public on(eventName: Events.kill, handler: (event?: KillEvent) => void): void;
public on(eventName: Events.initialize, handler: (event?: InitializeEvent) => void): void;
public on(eventName: Events.preupdate, handler: (event?: PreUpdateEvent) => void): void;
public on(eventName: Events.postupdate, handler: (event?: PostUpdateEvent) => void): void;
public on(eventName: Events.predraw, handler: (event?: PreDrawEvent) => void): void;
public on(eventName: Events.postdraw, handler: (event?: PostDrawEvent) => void): void;
public on(eventName: Events.predebugdraw, handler: (event?: PreDebugDrawEvent) => void): void;
public on(eventName: Events.postdebugdraw, handler: (event?: PostDebugDrawEvent) => void): void;
public on(eventName: Events.pointerup, handler: (event?: PointerEvent) => void): void;
public on(eventName: Events.pointerdown, handler: (event?: PointerEvent) => void): void;
public on(eventName: Events.pointermove, handler: (event?: PointerEvent) => void): void;
public on(eventName: Events.pointercancel, handler: (event?: PointerEvent) => void): void;
public on(eventName: string, handler: (event?: GameEvent<any>) => void): void;
public on(eventName: string, handler: (event?: GameEvent<any>) => void): void {
    this._checkForPointerOptIn(eventName);
    this.eventDispatcher.on(eventName, handler);
}
```